### PR TITLE
Decouple guava 25.0-jre dependency

### PR DESCRIPTION
**Pull request by anuchandy**
Removed azure-cosmos source dependency on the external guava-25.0-jre package.

Created a local package
com.azure.cosmos.implementation.guava25 that contains required guava types. The azure-cosmos source types are updated to use types from this package.

The non-shipping packages - sample, benchmark, and the test will continue to take a dependency on the external guava-25.0-jre package.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/pom.xml [hunk 1]
```diff
@@ -102,6 +102,19 @@ Licensed under the MIT License.
...
+         which is used in @Nullable annotation in reactor core classes.
+
+         Similar provided dependency exits for "azure-core" as well.
+    -->
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+      <version>3.0.2</version> <!-- {x-version-update;com.google.code.findbugs:jsr305;external_dependency} -->
+      <scope>provided</scope>
+    </dependency>
```
Comment (anuchandy) [inline sdk/cosmos/azure-cosmos/pom.xml]: azure-core reference: https://github.com/Azure/azure-sdk-for-java/blob/dea52be2efa93c5abecfdd0582b03a9e046cc957/sdk/core/azure-core/pom.xml#L73

**Comment by mbhaskar** [comment]
@anuchandy @JonathanGiles Is this process of copying classes from guava to our implementation a continuous one? I mean lets say we need a new class from guava in future, do we copy it then?

**Comment by JonathanGiles** [comment]
> @anuchandy @JonathanGiles Is this process of copying classes from guava to our implementation a continuous one? I mean lets say we need a new class from guava in future, do we copy it then?

No, my preference would be that we lessen our dependency, not strengthen it.

**Pull request merged**